src/main/java/org/myrobotlab/service/MqttBroker.java

```diff
@@ -0,0 +1,628 @@
+
+  public final static Logger log = LoggerFactory.getLogger(MqttBroker.class);
+
+  private static final long serialVersionUID = 1L;
+
+  private static String bytesToHex(byte[] hash) {
+    StringBuilder hexString = new StringBuilder(2 * hash.length);
+    for (int i = 0; i < hash.length; i++) {
+      String hex = Integer.toHexString(0xff & hash[i]);
+      if (hex.length() == 1) {
+        hexString.append('0');
+      }
+      hexString.append(hex);
+    }
+    return hexString.toString();
+  }
+
+  private static final MqttMsg toMqttMsg(InterceptMessage msg) {
+    MqttMsg retMsg = new MqttMsg();
+    if (msg instanceof InterceptPublishMessage) {
+      InterceptPublishMessage m = (InterceptPublishMessage) msg;
+      retMsg.setClientID(m.getClientID());
+      retMsg.setUsername(m.getUsername());
+      retMsg.setTopicName(m.getTopicName());
+      ByteBuf b = m.getPayload();
+      b.hasArray();
+      String payload = b.hasArray() ? new String(m.getPayload().array(), UTF_8) : b.toString(UTF_8);
+      retMsg.setPayload(payload);
+
+    } else if (msg instanceof InterceptAcknowledgedMessage) {
+      InterceptAcknowledgedMessage m = (InterceptAcknowledgedMessage) msg;
+      retMsg.setUsername(m.getUsername());
+
+    } else if (msg instanceof InterceptConnectionLostMessage) {
+      InterceptConnectionLostMessage m = (InterceptConnectionLostMessage) msg;
+      retMsg.setClientID(m.getClientID());
+      retMsg.setUsername(m.getUsername());
+
+    } else if (msg instanceof InterceptConnectMessage) {
+      InterceptConnectMessage m = (InterceptConnectMessage) msg;
+      retMsg.setClientID(m.getClientID());
+      retMsg.setUsername(m.getUsername());
+
+    } else if (msg instanceof InterceptDisconnectMessage) {
+      InterceptDisconnectMessage m = (InterceptDisconnectMessage) msg;
+      retMsg.setClientID(m.getClientID());
+      retMsg.setUsername(m.getUsername());
+
+    } else if (msg instanceof InterceptSubscribeMessage) {
+      InterceptSubscribeMessage m = (InterceptSubscribeMessage) msg;
+      retMsg.setClientID(m.getClientID());
+      retMsg.setUsername(m.getUsername());
+      retMsg.setTopicName(m.getTopicFilter());
+
+    } else if (msg instanceof InterceptUnsubscribeMessage) {
+      InterceptUnsubscribeMessage m = (InterceptUnsubscribeMessage) msg;
+      retMsg.setClientID(m.getClientID());
+      retMsg.setUsername(m.getUsername());
+      retMsg.setTopicName(m.getTopicFilter());
+    }
+    return retMsg;
+  }
+
+  protected String address = "0.0.0.0";
+
+  protected Boolean allow_anonymous = false;
+
+  protected Boolean allow_zero_byte_client_id = false;
+
+  protected Set<String> connectedClients = new HashSet<>();
+
+  /**
+   * determines if the broker will attempt to process any messages published on
+   * the api topics .. typically api/service & api/messages
+   */
+  boolean processApiMessages = true;
+
+  protected boolean listening = false;
+
+  transient final Server mqttBroker = new Server();

```

Comment:
does this need to be created here.. or perhaps it can be created in the startService method?  seems like that's the right place. but there may be other reasons to do it statically up when the class is loaded.

Reply:
your probably right ...

will analyze and fix

Reply:
it can't be final and get initialized in startService
The moquette broker is one of the few servers that can actually stop listening then start again successfully - so was going to utilize it - but I believe also the more defensive and safer way is to shut it down completely then new up a new one ...

Will investigate.

Reply:
I'm not going to change it ...
I might in the future make a new instance when listening and delete it when not listening ... but there should be some experimentation done around it's "internalPublish" which may not need to be listening to use ... dunno, don't want to mess with it until I do.